src/compiler/checker.ts

```diff
@@ -32023,14 +32045,46 @@ namespace ts {
         }
 
         function checkInExpression(left: Expression, right: Expression, leftType: Type, rightType: Type): Type {
-            if (leftType === silentNeverType || rightType === silentNeverType) {
-                return silentNeverType;
+            if (isPrivateIdentifier(left)) {
+                const lexicallyScopedSymbol = lookupSymbolForPrivateIdentifierDeclaration(left.escapedText, left);

```

Comment:
You should add a call to `checkExternalEmitHelpers` , and need to add a `ClassPrivateFieldIn` flag to `ExternalEmitHelpers` in types.ts. 

NOTE: in `checkExternalEmitHelpers` you *don't* need a branch like `ExternalEmitHelpers.ClassPrivateFieldGet` for `ClassPrivateFieldIn`. Those branches are checking for version-specific overloads from `tslib` which isn't needed in your case.

Reply:
done and added a test case for this.